### PR TITLE
chore(nov2023): updating to latest standards

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

- Follow items in https://mozilla-hub.atlassian.net/wiki/spaces/PE/pages/470582325/2024+-+Backend+Repo+Up+To+Date+Checklist
- Update node to node 20.10.0 to resolve memory leak issue https://mozilla-hub.atlassian.net/browse/POCKET-9193
- Add in sentry request isolation to make better work of Sentry errors
- Upgrade all packages and CI

## References

Documentation:
* Project doc

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -32,25 +32,27 @@ phases:
...
-      - cd .aws
-      - tfenv install
-      - npm ci
-      - npm run build
-      # synthesize the js into terraform json with the proper node environment
-      - 'if [ "$GIT_BRANCH" = "$DEV_BRANCH" ]; then NODE_ENV=development npm run synth; else npm run synth; fi'
-      - cd cdktf.out/stacks/list-api
-      - rm -rf .terraform # remove sym-link to ../.terraform (if this sym-link exists, terraform doesn't apply correctly. hmmmm)
-      - terraform init
+      - |
```
Comment (bassrock) [inline buildspec.yml:35]: Update to allow all the commands to run in the same shell environment for NVM to work

File: src/test/graphql/mutations/tagsMutationService-replaceSavedItemTags.integration.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ describe('tags mutation: replace savedItem tags', () => {
     clock = sinon.useFakeTimers({
       now: updateDate,
       shouldAdvanceTime: false,
+      shouldClearNativeTimers: true,
```
Comment (bassrock) [inline src/test/graphql/mutations/tagsMutationService-replaceSavedItemTags.integration.ts:39]: Warning produced without this on latest sinon

File: package.json [hunk 3]
```diff
@@ -8,86 +8,82 @@
     "build-schema": "node dist/server/buildSchema.js",
     "test-ci": "npm test",
     "test": "jest \"\\.spec\\.ts\"",
-    "test-integrations": "jest \"\\.integration\\.ts\" --runInBand",
+    "test-integrations": "jest \"\\.integration\\.ts\" --runInBand --forceExit",
```
Comment (bassrock) [inline package.json:11]: I had to add a forceExit. Theres a handle left open somewhere i could not find :( cc @kschelonka perhaps your wizardry knows where it is??

Comment (kschelonka) [reply package.json:11]: I will take a look!

**Review by kschelonka — APPROVED** [review]

**Pull request merged**